config/app.default.php

```diff
@@ -244,6 +249,8 @@
              * which is the recommended value in production environments
              */
             //'init' => ['SET GLOBAL innodb_stats_on_metadata = 0'],
+            
+            'url' => env('DATABASE_URL', ''),
```

Comment:
`env()` does some more things: http://book.cakephp.org/3.0/en/core-libraries/global-constants-and-functions.html

In case the environment variables come directly via $\_SERVER I don't see a problem (or rather, the problem is shifted to properly configuring the app server environment on the app server level). In case however there is a .env file things can clash. The convention should be helpful towards non-issues. The case you make above is already out of easy and conventional use.
If you don't like the `CAKE_` prefix, `CONF_` or `APP_` would do okay too. 

Another way to do it is to throw a HTTP 500 if a variable is set via `$_SERVER` and then is overwritten by .env. Hidden clashes are no fun (dangerous, error prone).